lib/matplotlib/backends/qt_compat.py

```diff
@@ -216,3 +220,49 @@ def _setDevicePixelRatio(obj, val):
     if hasattr(obj, 'setDevicePixelRatio'):
         # Not available on Qt4 or some older Qt5.
         obj.setDevicePixelRatio(val)
+
+
+class _allow_interrupt:
+    def __init__(self, qApp, old_sigint_handler):
+        self.interrupted_qobject = qApp
+        self.old_fd = None
+        if old_sigint_handler in (None, signal.SIG_IGN, signal.SIG_DFL):
```

Comment:
1) why do you want to not override SIG_DFL?
2) the first time you call signal.getsignal(SIGINT) (having not overriddent SIGINT before), you actually don't get SIG_DFL, but the private _signal.default_int_handler.  Perhaps that's an oversight in the docs, but in any case that case needs to be handled too (or SIG_DFL also needs to be overridable).

Reply:
Sorry for not answering for so long! This is the reason: https://github.com/matplotlib/matplotlib/pull/16311. SIG_DFL is used to indicate that a non-Python signal handler was installed for SIGINT. We have discussed with @tacaswell that this PR should not break #16311 and thus do not force new logic in that case.

Reply:
> SIG_DFL is used to indicate that a non-Python signal handler was installed for SIGINT.

I am confused: the docs (https://docs.python.org/3/library/signal.html#signal.getsignal) "signal.SIG_DFL means that the default way of handling the signal was previously in use, and None means that the previous signal handler was not installed from Python."

Reply:
Oh yes, I have confused that with the None case. I guess that is the ''knob'' to turn off the logic that @tacaswell has left. https://github.com/matplotlib/matplotlib/pull/13306#issuecomment-672509173 shows that, probably. So you first set up the DFL signal by
```Python
signal.signal(signal.SIGINT, signal.SIG_DFL)
```
and then the previous logic is restored.